site/docs/manual/Architecture.md

```diff
@@ -50,34 +50,77 @@ world = new World();
 
 ## Components
 A `Component` ([API Reference](/api/classes/component)) is an object that can store data but should have not behaviour (As that should be handled by systems). There is not a mandatory way to define a component.
-It could just be a function:
+
+To create a component you must extend the `Component` class and define a schema:
 ```javascript
-function ComponentA() {
-  this.number = 10;
-  this.string = "Hello";
+import { Component, Types } from 'ecsy';
+
+class ComponentA extends Component {}
+
+ComponentA.schema = {
+  number: { type: Types.Number, default: 10 },
+  string: { type: Types.String, default: "Hello" }
 }
 ```
 
-The recommended way is using ES6 classes and extending the `Component` class:
+The schema is used to set the default values of a component. ECSY also uses it to implement the default `.copy()`, `.clone()`, and `.reset()` methods. Schemas can also be used for tooling and serialization, something we plan on covering in the future.
+
+Each property in a schema represents a property on the component:
+
+```javascript
+const component = new ComponentA();
+console.log(component.number === 10); // true
+```
+
+The `type` field must be set for each property. Each type has a default value as well as `.copy()` and `.clone()` functions.
+
+ECSY comes with a few primitive types:
+- `Types.Number`: Defaults to `0`.
+- `Types.Boolean`: Defaults to `false`.
+- `Types.String`: Defaults to `""`.
+- `Types.Object`: Defaults to `undefined`. Copies by value, not a deep clone.
```

Comment:
having Array default to `[]` I'd expect Object to default to  `{}` why using undefined here instead?

Reply:
Maybe I should change Type.Object to Type.Value then? or Type.Ref? I'm often using Type.Object to define properties that are complex objects. If we wanted the default value to be `{}` we should at least do a shallow copy with `Object.assign(dest, src)`